app/ui-react/syndesis/src/app/UI.tsx

```diff
@@ -200,18 +199,24 @@ export const UI: React.FunctionComponent<IAppUIProps> = ({ routes }) => {
                                 'mailto:fuse-online-tech-preview@redhat.com';
                             }}
                             appNav={
-                              <AppTopMenu
-                                username={user.username || 'developer'}
-                              >
-                                <PfDropdownItem>
-                                  <Link
-                                    data-testid={'ui-logout'}
-                                    to={'/logout'}
-                                    className="pf-c-dropdown__menu-item"
-                                    children={t('Logout')}
-                                  />
-                                </PfDropdownItem>
-                              </AppTopMenu>
+                              <WithUserHelpers>
+                                {({ logout }) => {
+                                  return (
+                                    <AppTopMenu
+                                      username={user.username || 'developer'}
+                                      onSelectLogout={() => {
+                                        logout().then(() => {
+                                          console.log(
+                                            'do we need to manually redirect here?'
```

Comment:
I believe we need not do anything except resolve the promise really, the logout operation will make the browser navigate away from the app.